### PR TITLE
Add quick blog post about AudioContext outputLatency

### PR DESCRIPTION
**Pull request by beaufortfrancois**
@hoch Please have a look.

Note that I'm updating MDN browser compat data at https://github.com/mdn/browser-compat-data/pull/16426 in parallel

Live preview: https://deploy-preview-7991--web-dev-staging.netlify.app/audio-output-latency/

### CODE REVIEW

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Synchronize audio and video playback on the web
+subhead: >
+  A new Web Audio feature makes it possible to properly achieve AV-sync.
+date: 2022-05-25
+hero: image/vvhSqZboQoZZN9wBvoXq72wzGAf1/9nYDoT45VHvNAtTfLciY.jpeg
+alt: Flat screen TV turn on inside room photo
```
Comment (tomayac) [inline src/site/content/en/blog/audio-output-latency/index.md:7]: This looks broken, not sure what you wanted to say.

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - beaufortfrancois
+description: >
+  A new Web Audio feature makes it possible to properly achieve AV-sync.
+tags:
+  - blog
+  - audio
+  - media
+---
+
+The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware’s output latency (e.g. Bluetooth earbuds or USB external audio interface). This property is necessary to coordinate/compensate the latency from the input to the output. And it is also very useful when you need to synchronize a video stream and an audio stream produced by the [Web Audio API](https://developer.mozilla.org/docs/Web/API/Web_Audio_API). The process of synchronizing video and audio streams together is called [AV-sync](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```
Comment (tomayac) [inline src/site/content/en/blog/audio-output-latency/index.md:18]:
```suggestion
The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware's output latency (e.g. that of Bluetooth earbuds or of an external USB audio interface). This property is necessary to coordinate/compensate the latency from the input to the output. And it is also very useful when you need to synchronize a video stream and an audio stream produced by the [Web Audio API](https://developer.mozilla.org/docs/Web/API/Web_Audio_API). The process of synchronizing video and audio streams together is called [AV-sync](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```

**Review by tomayac — APPROVED** [review]
LGTM, just some drive-by suggestions.

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  - beaufortfrancois
+description: >
+  A new Web Audio feature makes it possible to properly achieve AV-sync.
+tags:
+  - blog
+  - audio
+  - media
+---
+
+The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware's output latency (e.g. that of Bluetooth earbuds or of an external USB audio interface). This property is necessary to coordinate/compensate the latency from the input to the output. And it is also very useful when you need to synchronize a video stream and an audio stream produced by the [Web Audio API](https://developer.mozilla.org/docs/Web/API/Web_Audio_API). The process of synchronizing video and audio streams together is called [AV-sync](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:18]:
```suggestion
The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware's output latency (for example, that of Bluetooth earbuds or of an external USB audio interface). This property is necessary to coordinate/compensate the latency from the input to the output. And it is also very useful when you need to synchronize a video stream and an audio stream produced by the [Web Audio API](https://developer.mozilla.org/docs/Web/API/Web_Audio_API). The process of synchronizing video and audio streams together is called [AV-sync](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+In [Chris Cunningham's demo](https://github.com/chcunningham/wc-talk#simple_video_playerhtml-live-demo), [WebCodecs](https://developer.mozilla.org/docs/Web/API/WebCodecs_API) is used to decode a `MediaStream` into raw video and audio data, and then played back into a HTML canvas element with audio data coming from the Web Audio API. The `outputLatency` property allows the demo to determine when a given audio timestamp is reaching the user's ears and then properly paint video frames to match that.
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:20]:
```suggestion
In [Chris Cunningham's demo](https://github.com/chcunningham/wc-talk#simple_video_playerhtml-live-demo), the [WebCodecs API](https://developer.mozilla.org/docs/Web/API/WebCodecs_API) is used to decode a `MediaStream` into raw video and audio data, and then played back into a HTML canvas element with audio data coming from the Web Audio API. The `outputLatency` property allows the demo to determine when a given audio timestamp is reaching the user's ears and then properly paint video frames to match that.
```

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+{% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/p61K2S4Wye34OJn4XZOj.png", alt="A screenshot of the demo", width="800", height="533" %}
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:22]:
```suggestion
<figure>
  {% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/p61K2S4Wye34OJn4XZOj.png", alt="A screenshot of the demo", width="800", height="533" %}
</figure>
```

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+Try it out for yourself, play the video with your favorite Bluetooth headset 🎧, wait for the bird 🐦 (see above), and toggle the checkbox ☑️ to observe audio playback changes. The total output latency value is updated in real-time.
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:24]:
```suggestion
Try it out for yourself, play the video with your favorite Bluetooth headset (🎧), wait for the bird (🐦) (see above), and toggle the checkbox (☑️) to observe audio playback changes. The total output latency value is updated in real-time.
```

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+### AudioContext outputLatency
+
+{% BrowserCompat 'api.AudioContext.outputLatency' %}
+
+Hero image by [Wahid Khene](https://unsplash.com/@wahidkhene) on [Unsplash](https://unsplash.com/photos/iKdQCIiSMlQ).
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:30]:
```suggestion
_Hero image by [Wahid Khene](https://unsplash.com/@wahidkhene) on [Unsplash](https://unsplash.com/photos/iKdQCIiSMlQ)._
```

**Review by malchata — APPROVED** [review]
Totally minor nits, otherwise LGTM! Apply/reject suggestions as needed and feel free to merge when you're ready.

**Comment by beaufortfrancois** [comment]
> Totally minor nits, otherwise LGTM! Apply/reject suggestions as needed and feel free to merge when you're ready.

Thank you @malchata!

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+---
+title: Synchronize audio and video playback on the web
+subhead: >
+  A new Web Audio feature makes it possible to properly achieve AV-sync.
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/audio-output-latency/index.md]: I've addressed your feedback. Good catch!

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  - beaufortfrancois
+description: >
+  A new Web Audio feature makes it possible to properly achieve AV-sync.
+tags:
+  - blog
+  - audio
+  - media
+---
+
+The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware's output latency (for example, that of Bluetooth earbuds or of an external USB audio interface). This property is necessary to coordinate/compensate the latency from the input to the output. And it is also very useful when you need to synchronize a video stream and an audio stream produced by the [Web Audio API](https://developer.mozilla.org/docs/Web/API/Web_Audio_API). The process of synchronizing video and audio streams together is called [AV synchronization](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```
Comment (hoch) [inline src/site/content/en/blog/audio-output-latency/index.md:18]: > This property is necessary to coordinate/compensate the latency from the input to the output.

Can we clarify this? The output latency can be useful when you want to:
- Synchronize the existing audio material and the newly recorded material. (in a music production scenario)
- Synchronize the Web Audio output and other media (e.g. video or MIDI playback).

So "coordinating from the input to the output" doesn't seem clear enough on what it does. Perhaps I am missing something.

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+In [Chris Cunningham's demo](https://github.com/chcunningham/wc-talk#simple_video_playerhtml-live-demo), the [WebCodecs API](https://developer.mozilla.org/docs/Web/API/WebCodecs_API) is used to decode a `MediaStream` into raw video and audio data, and then played back into a HTML `<canvas>` element with audio data coming from the Web Audio API. The `outputLatency` property allows the demo to determine when a given audio timestamp is reaching the user's ears and then properly paint video frames to match that.
```
[the remark this answers is not in the document]
Comment (hoch) [reply src/site/content/en/blog/audio-output-latency/index.md:20]: I was hoping to build a similar example under GoogleChromeLabs so we can have it as a reference implementation. But we can update the article when it is ready.

File: src/site/content/en/blog/audio-output-latency/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (hoch) [reply src/site/content/en/blog/audio-output-latency/index.md]: Correct. This is not new. Chrome is catching up the spec.

**Review by hoch — COMMENTED** [review]
Looks good! Only a few comments.

**Review by hoch — APPROVED** [review]
LGTM!

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  - audio
+  - media
+---
+
+The `outputLatency` property of an `AudioContext` instance provides an estimate of audio hardware's output latency (for example, that of Bluetooth earbuds or of an external USB audio interface). This property is useful when you want to:
+
+- Synchronize the existing audio material and the newly recorded material. (in a music production scenario)
+- Synchronize the [Web Audio](https://developer.mozilla.org/docs/Web/API/Web_Audio_API) output and other media (e.g. video or MIDI playback).
+
+The process of synchronizing video and audio streams together is called [AV synchronization](https://en.wikipedia.org/wiki/Audio-to-video_synchronization).
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/blog/audio-output-latency/index.md]: I've removed it.

**Comment by beaufortfrancois** [comment]
@malchata Feel free to have another look as I've made some modifications. Sorry for that.

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+- Synchronize the existing audio material and the newly recorded material. (in a music production scenario)
+- Synchronize the [Web Audio](https://developer.mozilla.org/docs/Web/API/Web_Audio_API) output and other media (e.g. video or MIDI playback).
+
+In this [WebCodecs demo](https://wc-talk.netlify.app/simple_video_player.html) ([source](https://github.com/chcunningham/wc-talk)), the [WebCodecs API](https://developer.mozilla.org/docs/Web/API/WebCodecs_API) is used to decode a `MediaStream` into raw video and audio data, and then played back into a HTML `<canvas>` element with audio data coming from an [Audio Worklet](https://developer.mozilla.org/docs/Web/API/AudioWorklet). The `outputLatency` property allows the demo to determine when a given audio timestamp is reaching the user's ears and then properly paint video frames to match that.
+
+<figure>
+  {% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/p61K2S4Wye34OJn4XZOj.png", alt="A screenshot of the demo", width="800", height="533" %}
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:26]:
```suggestion
  {% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/p61K2S4Wye34OJn4XZOj.png", alt="A screenshot of the demo, which is of an embedded video with audio controls for volume, audio buffer health, total output latency, and a checkbox to use AudioContext.outputLatency.", width="800", height="533" %}
```

**Comment by malchata** [comment]
@beaufortfrancois, one minor nit to commit, but LGTM. Merge when ready.

File: src/site/content/en/blog/audio-output-latency/index.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
+---
+title: Synchronize audio and video playback on the web
+subhead: >
+  The Web Audio API makes it possible to properly achieve AV synchronization.
+date: 2022-05-25
```
Comment (malchata) [inline src/site/content/en/blog/audio-output-latency/index.md:5]:
```suggestion
date: 2022-05-27
```

**Comment by beaufortfrancois** [comment]
I'll wait for https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/outputLatency to be updated before merging. Thank you @malchata

**Pull request merged**